File: LegendEditor/Assets/Scripts/Game/DataBase/SQLiteDataBase/SQLiteClothFrame.cs
using Mono.Data.Sqlite;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

public class ClothFrameData
{
	public int mID;
	public string mLabel;
	public int mDirection;
	public string mAction;
	public int mFrameCount;
	public List<int> mPosX;
	public List<int> mPosY;
}

public class SQLiteClothFrame : SQLiteTable
{
	string COL_ID = "ClothID";
	string COL_LABEL = "Label";
	string COL_DIRECTION = "Direction";
	string COL_ACTION = "Action";
	string COL_FRAME_COUNT = "FrameCount";
	string COL_POSX = "PosX";
	string COL_POSY = "PosY";
	public SQLiteClothFrame(SQLite sqlite)
		:base("ClothFrame", sqlite)
	{
		;
	}
	public void query(int id, out List<ClothFrameData> dataList)
	{
		string condition = "";
		appendConditionInt(ref condition, COL_ID, id, "");
		parseReader(doQuery(condition), out dataList);
	}
	public void query(int id, int direction, out List<ClothFrameData> dataList)
	{
		string condition = "";
		appendConditionInt(ref condition, COL_ID, id, " and ");
		appendConditionInt(ref condition, COL_DIRECTION, direction, "");
		parseReader(doQuery(condition), out dataList);
	}
	public void query(int id, int direction, string action, out List<ClothFrameData> dataList)
	{
		string condition = "";
		appendConditionInt(ref condition, COL_ID, id, " and ");
		appendConditionInt(ref condition, COL_DIRECTION, direction, " and ");
		appendConditionString(ref condition, COL_ACTION, action, "");
		parseReader(doQuery(condition), out dataList);
	}
	public void updateData(ClothFrameData data)
	{
		string valueString = "";
		appendUpdateInt(ref valueString, COL_FRAME_COUNT, data.mFrameCount);
		appendUpdateIntArray(ref valueString, COL_POSX, data.mPosX);
		appendUpdateIntArray(ref valueString, COL_POSY, data.mPosY);
		string condition = "";
		appendConditionInt(ref condition, COL_ID, data.mID, " and ");
		appendConditionInt(ref condition, COL_DIRECTION, data.mDirection, " and ");
		appendConditionString(ref condition, COL_ACTION, data.mAction, "");
		doUpdate(valueString, condition);
	}
	public void insert(ClothFrameData data)
	{
		string valueString = "";
		appendValueInt(ref valueString, data.mID);
		appendValueString(ref valueString, data.mLabel);
		appendValueInt(ref valueString, data.mDirection);
		appendValueString(ref valueString, data.mAction);
		appendValueInt(ref valueString, data.mFrameCount);
		appendValueIntArray(ref valueString, data.mPosX);
		appendValueIntArray(ref valueString, data.mPosY, true);
		doInsert(valueString);
	}
//------------------------------------------------------------------------------------------------------------------------
protected void parseReader(SqliteDataReader reader, out List<ClothFrameData> dataList)
	{
		dataList = new List<ClothFrameData>();
		while (reader.Read())
		{
			ClothFrameData data = new ClothFrameData();
			data.mID = StringUtility.stringToInt(reader[COL_ID].ToString());
			data.mLabel = reader[COL_LABEL].ToString();
			data.mDirection = StringUtility.stringToInt(reader[COL_DIRECTION].ToString());
			data.mAction = reader[COL_ACTION].ToString();
			data.mFrameCount = StringUtility.stringToInt(reader[COL_FRAME_COUNT].ToString());
			StringUtility.stringToIntArray(reader[COL_POSX].ToString(), ref data.mPosX);
			StringUtility.stringToIntArray(reader[COL_POSY].ToString(), ref data.mPosY);
			dataList.Add(data);
		}
		reader.Close();
	}
};

File: LegendDataEditor/LegendDataEditor/Sqlite/Table/SQLiteClothFrame.cpp
#include "SQLiteClothFrame.h"
#include "Utility.h"
#include "SQLiteDataReader.h"
#include "SQLite.h"

std::string ClothFrameData::COL_ID = "ClothID";
std::string ClothFrameData::COL_LABEL = "Label";
std::string ClothFrameData::COL_DIRECTION = "Direction";
std::string ClothFrameData::COL_ACTION = "Action";
std::string ClothFrameData::COL_FRAME_COUNT = "FrameCount";
std::string ClothFrameData::COL_POSX = "PosX";
std::string ClothFrameData::COL_POSY = "PosY";

void SQLiteClothFrame::query(int clothID, txVector<ClothFrameData>& dataList)
{
	std::string conditionString;
	appendConditionInt(conditionString, ClothFrameData::COL_ID, clothID, "");
	std::string queryStr = "SELECT * FROM " + mTableName + " WHERE " + conditionString;
	parseReader(mSQLite->executeQuery(queryStr), dataList);
}
void SQLiteClothFrame::query(int clothID, int direction, txVector<ClothFrameData>& dataList)
{
	std::string conditionString;
	appendConditionInt(conditionString, ClothFrameData::COL_ID, clothID, " and ");
	appendConditionInt(conditionString, ClothFrameData::COL_DIRECTION, direction, "");
	std::string queryStr = "SELECT * FROM " + mTableName + " WHERE " + conditionString;
	parseReader(mSQLite->executeQuery(queryStr), dataList);
}
void SQLiteClothFrame::query(int clothID, int direction, const std::string& action, txVector<ClothFrameData>& dataList)
{
	std::string conditionString;
	appendConditionInt(conditionString, ClothFrameData::COL_ID, clothID, " and ");
	appendConditionInt(conditionString, ClothFrameData::COL_DIRECTION, direction, " and ");
	appendConditionString(conditionString, ClothFrameData::COL_ACTION, action, "");
	std::string queryStr = "SELECT * FROM " + mTableName + " WHERE " + conditionString;
	parseReader(mSQLite->executeQuery(queryStr), dataList);
}
bool SQLiteClothFrame::updateData(const ClothFrameData& data)
{
	std::string updateString;
	appendUpdateInt(updateString, ClothFrameData::COL_FRAME_COUNT, data.mFrameCount);
	appendUpdateIntArray(updateString, ClothFrameData::COL_POSX, data.mPosX);
	appendUpdateIntArray(updateString, ClothFrameData::COL_POSY, data.mPosY, true);
	std::string conditionString;
	appendConditionInt(conditionString, ClothFrameData::COL_ID, data.mID, " and ");
	appendConditionInt(conditionString, ClothFrameData::COL_DIRECTION, data.mDirection, " and ");
	appendConditionString(conditionString, ClothFrameData::COL_ACTION, data.mAction, "");
	return doUpdate(updateString, conditionString);
}
bool SQLiteClothFrame::insert(const ClothFrameData& data)
{
	std::string valueString;
	appendValueInt(valueString, data.mID);
	appendValueString(valueString, data.mLabel);
	appendValueInt(valueString, data.mDirection);
	appendValueString(valueString, data.mAction);
	appendValueInt(valueString, data.mFrameCount);
	appendValueIntArray(valueString, data.mPosX);
	appendValueIntArray(valueString, data.mPosY, true);
	return doInsert(valueString);
}